src/atlas/interpolation/method/cubedsphere/CubedSphereBilinear.cc

```diff
@@ -33,6 +33,13 @@ void CubedSphereBilinear::do_setup(const FunctionSpace& source, const FunctionSp
     ATLAS_ASSERT(ncSource);
     ATLAS_ASSERT(target_);
 
+    // Add tile index metadata to target.
+    target_->metadata().set("tile index", std::vector<idx_t>{});
```

Comment:
It doesn't look right that we're modifying the target function space metadata here, especially since it should be `const`.

Reply:
This should be changed downstream:
```
std::vector<idx_t> tile_index;
target_fs.metadata().get("tile index", tile_index);
```

Reply:
Fair. I was being lazy because it's already being set on line 109 of this file. I can remove it. Two wrongs don't make a right.

I plan on removing this and the `allow_halo_exchange_` boolean once the oops PRs are merged. It's terribly hacky.

Reply:
So you agree to modify this PR to correct the wrongs, and leave this hanging until oops PRs are fixed? I'm not entirely clear on your intention.

Reply:
Sorry, I'm communicating poorly.

This PR:
I'll revert the metadata change on line 37.

Next one (once oops PRs are settled):
I'll remove the line 109 metadata and the halo exchange control. This will involve a significant amount of downstream changes, so It'll take a bit of time.